testutil/gen_app.go

```diff
@@ -4,37 +4,44 @@ import (
 	"time"
 
 	sdkmath "cosmossdk.io/math"
+	dbm "github.com/cometbft/cometbft-db"
+	"github.com/cometbft/cometbft/libs/log"
+	tmtypes "github.com/cometbft/cometbft/types"
+	"github.com/cosmos/cosmos-sdk/baseapp"
 	"github.com/cosmos/cosmos-sdk/codec"
 	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
 	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
 	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
-	"github.com/cosmos/cosmos-sdk/simapp"
 	"github.com/cosmos/cosmos-sdk/testutil/mock"
+	simtestutil "github.com/cosmos/cosmos-sdk/testutil/sims"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
 	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
 	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
-	"github.com/tendermint/tendermint/libs/log"
-	tmtypes "github.com/tendermint/tendermint/types"
-	dbm "github.com/tendermint/tm-db"
+
+	"github.com/ignite/modules/app"
 
 	spnapp "github.com/tendermint/spn/app"
 	"github.com/tendermint/spn/cmd"
 )
 
-func GenApp(withGenesis bool, invCheckPeriod uint) (*spnapp.App, spnapp.GenesisState) {
-	db := dbm.NewMemDB()
-	encCdc := cmd.MakeEncodingConfig(spnapp.ModuleBasics)
-	app := spnapp.New(
-		log.NewNopLogger(),
-		db,
-		nil,
-		true,
-		map[int64]bool{},
-		simapp.DefaultNodeHome,
-		invCheckPeriod,
-		encCdc,
-		simapp.EmptyAppOptions{})
+func GenApp(chainID string, withGenesis bool, invCheckPeriod uint) (*spnapp.App, spnapp.GenesisState) {
+	var (
+		db     = dbm.NewMemDB()
+		encCdc = cmd.MakeEncodingConfig(app.ModuleBasics)
+		app    = spnapp.New(
+			log.NewNopLogger(),
+			db,
+			nil,
+			true,
+			map[int64]bool{},
+			app.DefaultNodeHome,
+			invCheckPeriod,
+			encCdc,
+			simtestutil.EmptyAppOptions{},
+			baseapp.SetChainID(chainID),
+		)
+	)
```

Comment:
This change should be applied if the comment above is applied

```suggestion
		encCdc = cmd.MakeEncodingConfig(spnapp.ModuleBasics)
		app    = spnapp.New(
			log.NewNopLogger(),
			db,
			nil,
			true,
			map[int64]bool{},
			spnapp.DefaultNodeHome,
			invCheckPeriod,
			encCdc,
			simtestutil.EmptyAppOptions{},
			baseapp.SetChainID(chainID),
		)
	)
```